### ISSUE TITLE
Inconsistent nowplaying API content on live stream

### ISSUE THREAD

**Issue by gammaw** [AzuraCast/AzuraCast#1968]
Issue: Azura will remember the live stream's text from the last session and output it in the nowplaying API. I would expect an empty string there as long as the user (Butt) won't send a custom text. 

Here is an example. A song is going on, live is false, everything is alright. This is the relevant part of the nowplaying API: 

```
"live": {
        "is_live": false,
        "streamer_name": ""
    },
    "now_playing": {
        "song": {
            "artist": "DJ Best",
            "title": "BestSong",
        }
    }
```

Next, a live DJ connects via Butt, this is how the API will look like:  

```
"live": {
        "is_live": true,
        "streamer_name": "Best Live DJ"
    },
    "now_playing": {
        "song": {
            "artist": "",
            "title": "something_from_the_past",
        }
    }
```

I can change `something_from_the_past` by sending a message from Butt to Azura via `Update song name manually` in Butt. But 1) the user on the frontend will see a wrong text in the meanwhile, 2) DJs don't always want to have a title. 

Would it be possible that Azuracast initiates `title` upon new connecting live streams? It would be even better if a new field would be added to the `live `part. Would that be possible?

**Comment by Vaalyn** [comment]
This issue seems to be a duplicate of #867, please refer to that thread for further updates regarding the issue of blank metadata when live streaming.

**Comment by gammaw** [comment]
@Vaalyn are you sure? It doesn't seem to be an issue with the streaming client (Butt). `something_from_the_past` is sent by Azuracast not by Butt. I know it because in our experiments  `something_from_the_past` could not be sent by Butt because we used a different Butt client which had never sent `something_from_the_past` before. So it must come from Azuracast's "memory". That's why I wondered if the initialisation on every new live DJ connection could be extended by emptying this string in the JSON message?

**Comment by gammaw** [comment]
@SlvrEagle23 @Vaalyn this issue was closed and marked as duplicate with #867. Like I said in my last comment back in September it doesn't seem to me to be an issue with Butt. We're having this issue still and since our main tool for live shows is Butt, it's a very important issue for us. I didn't want to open a new issue, since this one describes our issue in details. 

Again, the problem is that our live show hosts (using Butt) must set a title (in Butt via "Update song name manually") otherwise AzuraCast will send the last title used. Can't you set an empty string when a new Butt source connects? The client can then update it on demand, otherwise, it remains empty. What do you think?

**Comment by gammaw** [comment]
@SlvrEagle23 can you tell me what exactly happens when Butt sends a new song name and when a similar entry appears in the Liquidsoap log?

`2020/02/16 19:40:47 [radio_input_streamer:3] New metadata chunk ? -- some_text.`

I would like to know how Azuracast will learn about this update and how it will update `some_text` for the song title in the nowplaying JSON message. Can you point me to the code? 

In theory, I could force an update, for example in the `djon` handler code, to quick fix this issue.

**Comment by gammaw** [comment]
My gap in understanding is how the metadata chunk update (from Butt to Liquidsoap) is communicated toward Azuracast. I can see in the Liquidsoap configuration that every metadata change (upon new a song) triggers a call to `InternalController.feedbackAction()` but not in case of the metadata chunk update. Is it done pulling-wise via one of the synch tasks in Azuracast? I appreciate your help.

**Comment by gammaw** [comment]
FYI: after studying the Azuracast code base, many things became clear (many not yet). The original issue (inconsistent nowplaying data) seems to be caused by Liquidsoap (not Azuracast). I tried a solution similar to @SlvrEagle23's [reverted commit](https://github.com/AzuraCast/AzuraCast/commit/44f8f806d0d7967b6816ed0271f4592781cce0da) only that I inserted the metadata directly in Liquidsoap instead of via telnet. And then I found out that Liquidsoap overwrites my insertion with metadata from the last previous live session. It can be clearly seen from the Icecast access log. I created an upstream issue (https://github.com/savonet/liquidsoap/issues/1096). I still don't understand how Azuracast learns about these metadata changes but it does (I see the change propagated into the nowplaying data).

**Comment by Vaalyn** [comment]
@gammaw regarding how AC receives metadata changes, take a look at the generated liquidsoap config. Specifically this part:

```
# Send metadata changes back to AzuraCast
def metadata_updated(m) =
  if (m["song_id"] != "") then
    ret = list.hd(get_process_lines("curl -s --request POST --url http://web/api/internal/1/feedback --form song="^string.quote(m["song_id"])^" --form media="^string.quote(m["media_id"])^" --form playlist="^string.quote(m["playlist_id"])^" --form api_auth="^string.quote("xxxxxxxx")^""), default="")
    log("AzuraCast Feedback Response: #{ret}")
  end
end

radio = on_metadata(metadata_updated,radio)
```

This calls the [InternalController](https://github.com/AzuraCast/AzuraCast/blob/master/src/Controller/Api/InternalController.php#L146) on metadata changes.

**Comment by gammaw** [comment]
@Vaalyn thanks, I know that part. My question is what if the `m["song_id"] == ""` (no feedback is sent to AC)?  Because this is the case when Butt sends metadata in the live stream.

**Comment by gammaw** [comment]
I figured it out, finally. AC periodically queries the `/admin/stats` interface of the Icecast broadcast stream ([this](https://github.com/AzuraCast/nowplaying/blob/master/src/Adapter/Icecast.php) is the responsible PHP class). The response contains the current metadata of the live stream between Butt and Liquidsoap. 

For me the solution of this issue is outside AC and could be solved like this: 
- make Liquidsoap not send the previous metadata when a live source connects (= disable `replay_metadata` for the live source - see https://github.com/savonet/liquidsoap/issues/1096), and
- make Butt not send the previously set song metadata upon new connection (I emailed the author).
